tests/CommunityToolkit.Aspire.Hosting.GoFeatureFlag.Tests/GoFeatureFlagFunctionalTests.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Polly;
+using Xunit.Abstractions;
+using Aspire.Hosting.Utils;
+using CommunityToolkit.Aspire.Testing;
+using OpenFeature.Contrib.Providers.GOFeatureFlag;
+using OpenFeature.Model;
+using YamlDotNet.Core.Tokens;
+
+namespace CommunityToolkit.Aspire.Hosting.GoFeatureFlag.Tests;
+
+[RequiresDocker]
+public class GoFeatureFlagFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    public async Task VerifyGoFeatureFlagResource()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create(testOutputHelper);
+
+        var goff = builder.AddGoFeatureFlag("goff");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+#pragma warning disable CTASPIRE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        await app.WaitForTextAsync("GO Feature Flag Relay Proxy", goff.Resource.Name);
```

Comment:
Since it's using a health check, we should be able to wait for the resource to be ready rather than the hacky way of reading the logs.